americangut/notebook_environment.py

```diff
@@ -0,0 +1,357 @@
+
+    if find_executable('mod2_pcoa.py') is None:
+        raise EnvironmentError("The AG scripts are not in $PATH.")
+_assert_environment()
+
+
+def activate(chp):
+    """Activate a chapter
+
+    Parameters
+    ----------
+    chp : str
+        The chapter.
+
+    Returns
+    -------
+    str
+        The path to the activated directory
+
+    Notes
+    -----
+    Activation creates the chapter processing directory if it does
+    not already exist.
+    """
+    path = os.path.join(ag.working_dir, chp)
+    if not os.path.exists(path):
+        os.mkdir(path)
+    return path
+
+
+def get_sortmerna_index():
+    """Return the absolute path a SortMeRNA index if available"""
+    return os.environ.get('AG_SMR_INDEX')
+
+
+def get_rarefaction_depth():
+    """Return the rarefaction depth to use"""
+    if _TEST_ENV:
+        return "100"
+    else:
+        return "1000"
+
+
+def get_reference_set():
+    """Get the reference set to use for OTU picking
+
+    Returns
+    -------
+    (str, str)
+        The file paths to the reference sequences and the reference taxonomy.
+    """
+    if _TEST_ENV:
+        repo = get_repository_dir()
+        ref_seqs = os.path.join(repo, 'tests/data/otus.fna')
+        ref_tax = os.path.join(repo, 'tests/data/otus.txt')
+        return ref_seqs, ref_tax
+    else:
+        return qdr.get_reference_sequences(), qdr.get_reference_taxonomy()
+
+
+def get_hmp():
+    """Get the HMP 100nt table and mapping"""
+    return _get_data('HMP', 'HMPv35_100nt')
+
+
+def get_pgp():
+    """Get the PGP 100nt table and mapping"""
+    return _get_data('PGP', 'PGP_100nt')
+
+
+def get_global_gut():
+    """Get the Global Gut table and mapping"""
+    return _get_data('GG', 'GG_100nt')
+
+
+def _get_data(data_dir, tag):
+    """Get a non-AG table and mapping file
+
+    Parameters
+    ----------
+    data_dir : str
+        The base data path
+    tag : str
+        The filetag (e.g., HMPv35_100nt)
+
+    Notes
+    -----
+    If $AG_TESTING == 'True', then the data returned will correspond to the
+    test dataset.
+
+    Raises
+    ------
+    IOError
+        If the filepaths are not accessible
+
+    Returns
+    -------
+    (str, str)
+        The filepath to the table, and the filepath to the mapping file.
+    """
+    repo = get_repository_dir()
+    data = 'tests/data' if _TEST_ENV else 'data'
+    base = os.path.join(repo, data)
+
+    table = os.path.join(base, data_dir, '%s.biom' % tag)
+    mapping = os.path.join(base, data_dir, '%s.txt' % tag)
+
+    if not os.path.exists(table):
+        raise IOError("Unable to access: %s" % table)
+    if not os.path.exists(mapping):
+        raise IOError("Unable to access: %s" % table)
+
+    return table, mapping
+
+
+def get_accessions():
+    """Get the accessions to use, or redirect to test data
+
+    Notes
+    -----
+    If $AG_TESTING == 'True', then the accessions returned will
+    correspond to the test dataset.
+
+    Returns
```

Comment:
There's an order of information given in [the howto](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections) but not actually explicit in whether that order must be maintained.